File: in_Python/0448 Find All Numbers Disappeared in an Array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = []
        for i in range(len(nums)):
            idx = abs(nums[i]) - 1
            nums[idx] = - abs(nums[idx])
        for i in range(len(nums)):
            if nums[i] > 0:
                res.append(i+1)
        return res
            
        

File: in_Java/0448 Find All Numbers Disappeared in an Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList<>();
        for(int i = 0; i < nums.length; i++){
            int cur = Math.abs(nums[i]);
            //if(cur == 0) continue;
            if(nums[cur-1] > 0) nums[cur-1] = -nums[cur-1];
        }
        for(int i = 0; i < nums.length; i++){
            if(nums[i] > 0) res.add(i+1);
        }
        return res;
    }
}